yoyodyne/data/indexes.py

```diff
@@ -42,9 +42,7 @@ def __init__(
         # If embeddings are tied, so are the vocab items.
         # Then, the source and target vocabularies are the union.
         if self.tie_embeddings:
-            vocabulary = list(
-                sorted(set(source_vocabulary) | set(target_vocabulary))
-            )
+            vocabulary = sorted(set(source_vocabulary) | set(target_vocabulary))
```

Comment:
Oh, I may have mispoke here earlier. This won't work because you're unioning the sets first. I went off half-cocked and what you had earlier is actually correct. Sorry about that.